captura/views.py

```diff
@@ -224,7 +227,121 @@ def create_estudio(request):
     if request.method == 'POST':
         familia = Familia.objects.create()
         Estudio.objects.create(capturista=request.user.capturista, familia=familia)
-        return redirect(reverse('home'))
+        return redirect(reverse('captura:familia', kwargs={'id_familia': familia.id}))
+
+
+@login_required
+@user_passes_test(is_capturista)
+def familia(request, id_familia):
+    """ This view allows a capturista to capture the information related
+    to a specific family.
+    """
+    if request.method == 'POST':
+        instance = get_object_or_404(Familia, pk=id_familia)
+        form = FamiliaForm(request.POST, instance=instance)
+        if form.is_valid():
+            form.save()
+            return redirect(reverse('captura:integrantes',
+                                    kwargs={'id_familia': form.instance.pk}))
+    else:
+        familia = Familia.objects.get(pk=id_familia)
+        form = FamiliaForm(instance=familia)
+        form_view = 'captura/familia_form.html'
+        return render(request, 'captura/captura_base.html', {'form_view': form_view,
+                                                             'form': form,
+                                                             'familia': familia})
+
+
+@login_required
+@user_passes_test(is_capturista)
+def integrantes(request, id_familia):
+    """ This view allows a capturista to see all the information about the
+    integrantes of a specific family, they are displayed inside a table,
+    and the capturista can select each of them individually.
+    """
+    context = {}
+
+    integrantes = Integrante.objects.filter(familia__pk=id_familia)
+    familia = Familia.objects.get(pk=id_familia)
+    context['integrantes'] = integrantes
+    context['familia'] = familia
+    return render(request, 'captura/dashboard_integrantes.html', context)
+
+
+def integrante(request, id_integrante):
+    """ This view allows, the users to edit the information of a family memeber.
+    """
+    tutores = ['Tutor', 'Madre', 'Padre']
+    if request.method == 'POST':
+        instance = get_object_or_404(Integrante, pk=id_integrante)
+        integrante_form = IntegranteForm(request.POST, instance=instance)
+        if integrante_form.is_valid():
```

Comment:
the custom validator for phone is not working properly. im not sure why but if the phone is invalid it returns None